_RFCs/0007-deprecating-packages.md

```diff
@@ -0,0 +1,65 @@
+* [core/jre9](https://github.com/habitat-sh/core-plans/tree/master/jre9)(Oracle has indicated [here](http://www.oracle.com/technetwork/java/javase/downloads/jre9-downloads-3848532.html) that Java SE 9 has reached end of support and users should upgrade to Java SE 10. Additionally, Oracle's download links to JRE 9 and JDK 9 have been removed. It should be noted that Oracle DOES still support JRE 8 and JDK 8.)
+* [core/jdk9](https://github.com/habitat-sh/core-plans/blob/master/jdk9/plan.sh)(see explanation for core/jre9)
+
+When we become aware that a piece of software we package in core plans has been EOL'd or the author/manufacturer otherwise stops supporting it, we will:
+* Open an RFC explaining why the plan must be deprecated
+* When that RFC is approved (though the normal RFC approval process), we must
+1) Delist the package in builder by setting it to "private" (that will make it invisible to anyone outside of the core org, as mentioned [in this issue](https://github.com/habitat-sh/builder/issues/18)
+2) Add a "Deprecated" notice to the core plan's README in the [core plans repo](https://github.com/habitat-sh/core-plans)
+3) Add some sort of indication to the core plan's view page in Builder (remember, even if it is set to private and not viewable publicly, it will still be viewable to members of the core origin) (this will likely require some feature work)
+4) Remove the plan from the [.bldr.toml](https://github.com/habitat-sh/core-plans/blob/master/.bldr.toml) file in the core plans repo.
+5) Disconnect the plan from Builder
+
+We must follow this same process in the case of a plan that
+* is incorrectly scoped into core plans - i.e. a test package
+* or has been moved out to a different origin - i.e. builder packages were removed from the core origin earlier this year
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+See the Guide-level explanation
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This would necessitate users changing their own behavior and the behavior of their software.
+
+It will also necessitate some feature work on the part of Builder - we would need a field of something to that effect indicating when a plan is deprecated.
+
+# Rationale and alternatives
+[alternatives]: #alternatives
+
+Here are few alternatives:
+
+* Removing unsupported software completely from Builder - this would break anyone who depends on the software and cannot upgrade at this time. If there is at least one successful build of the software, we should preserve it on Builder for those who must use it, while indicating that it is deprecated and users should upgrade as soon as they can. Setting the plan as "private" makes the plan unsearchable, but still downloadable by those who are already using it.
```

Comment:
We should consider cases where removal of packages is allowable.  For example, we have 'test' packages in core that can be safely removed even though they have other (also testing) packages that depend on them.  Another example - an accidental release of sensitive or private info.  There may be other cases where legally we are obligated to do so.  Having a blanket policy where nothing is ever removed seems too inflexible.

Reply:
I'm wholly in agreement with salim here. I'd like to have some way to do this, but I also don't think that necessarily prohibits me from giving this a :+1: